pkg/k8s/watchers/watcher.go

```diff
@@ -391,7 +395,17 @@ func (k *K8sWatcher) GetCachedPod(namespace, name string) (*slim_corev1.Pod, err
 }
 
 // GetCachedNamespace returns a namespace from the local store.
+// If network policies are disabled, labels are not needed, the namespace
+// watcher is not running, and a namespace containing only the name is returned.
 func (k *K8sWatcher) GetCachedNamespace(namespace string) (*slim_corev1.Namespace, error) {
+	if !option.NetworkPolicyEnabled(option.Config) {
```

Comment:
This strikes me as a huge source of potential bugs.

Can we just get rid of this function entirely?

Reply:
I pushed it up to the `FetchNamespace` func.

Currently, we cannot remove it, because specific watchers (e.g. namespace watcher) are not accessible via `K8sWatcher` struct.
We would need to refactor the `K8sWatcher` struct.
We want to do eventually remove the watchers there, and enable all cells via `resource.Resource[]`.
It's out of scope of this change.